plugins/modules/ec2_vpc_igw.py

```diff
@@ -218,6 +220,11 @@ def ensure_igw_present(self, vpc_id, tags, purge_tags):
             resource_type='internet-gateway', tags=tags, purge_tags=purge_tags
         )
         igw = self.get_matching_igw(vpc_id)
+
+        if igw is None:
+            self._module.fail_json(
+                msg='EC2 could not find an Internet Gateway attached to VPC {0}, aborting'.format(vpc_id))
+
```

Comment:
Let's try again with a waiter.

Reply:
updated with manual waiter. seems to be the best solution after failing with traditional waiter. the issue is inconsistency in calling describe_internet_gateways

Reply:
@jatorcasso Kudos for solving this, we used to do a lot of functions like this and the solution does work.  We've since added a module utility though to extend botocore's waiters that you can use in cases like this: https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/waiters.py

Reply:
@jillr so I first tried to add a waiter there that waits for the attachment to be `available`, but it still errors out because the issue is sometimes the describe method cant find a gateway thats present and successfully attached to a vpc

Reply:
so to clarify, calling the waiter from module_util like below is still running into NoneType errors